protocol/communities/community_test.go

```diff
@@ -483,18 +483,30 @@ func (s *CommunitySuite) TestValidateRequestToJoin() {
 	}
 }
 
+//FIXME: Revise me, ideally split expand into matrix below
+// Community Permission - Membership - Grant - Can Post
+// MEMBERSHIP-NO-MEMBERSHIP-Member-> User can post
+// MEMBERSHIP-NO-MEMBERSHIP->NON member -> User can't post
+// MEMBERSHIP-NO-MEMBERSHIP-Grant -> user can post
+// MEMBERSHIP-NO-MEMBERSHIP-old-grant -> user can't post
+
+//FIXME: add complete permissions matrix
+// | Community Perm | Chat Perm | Chat Member | Grant   | Can Post |
+// |----------------|-----------|-------------|---------|----------|
+// | MEMBERSHIP     | x         | Yes         | -       | Yes      |
+// | MEMBERSHIP     | x         | No          | -       | No       |
+// | NO-MEMBERSHIP  | x         | Yes         | -       | Yes      |
+// | NO-MEMBERSHIP  | x         | No          | -       | No       |
+// | NO-MEMBERSHIP  | x         | No          | Valid   | Yes?     |
+// | NO-MEMBERSHIP  | x         | No          | Invalid | No       |
```

Comment:
@cammellos can you shortly explain what a grant is? I feel like it's probably something from an old requirement, but I don't remember seeing it in any of the new requirements or code.

I'm of the opinion that if someone is not part of the member list, they shouldn't be able to post ever. However, this test below seems to imply that with a grant, it should be possible. Is that still something we want? If not, might as well remove all the grant code I guess?

cc @John-44 to clarify requirements. Basically, am I correct that you 100% need to be a member of a community/chat to be able to post to it?

Reply:
I just discussed this with @cammellos and he said:
> so, the idea is that when you join a community, you get a "grant"
and that's a proof you have been added
so when you send a message, you send also the grant
so other clients can optimistically process the message, if the community description hasn't reached them yet

But it turns out we never fully supported it anyway:
> we never really supported it

He then told me it's fine to remove it.

I'll open an issue in a few.

@shinnok that means you can just set the grant tests to `canPost: false`, so that the test will pass. Then someone will have the chore of removing the grant code that's no longer useful later.

Reply:
I added the issue to remove the grant code here: https://github.com/status-im/status-go/issues/4180

Reply:
I marked the sub-tests that are supposed to fail in TestCanPost().